openlibrary/core/models.py

```diff
@@ -593,6 +596,108 @@ def get_redirect_chain(work_key: str) -> list:
                 resolved_key = thing.key
         return redirect_chain
 
+    @classmethod
+    def resolve_redirect_chain(cls, work_key: str, test: bool=False):
+        summary = {
+            'key': work_key,
+            'redirect_chain': [],
+            'resolved_key': None
+        }
+        redirect_chain = cls.get_redirect_chain(work_key)
+        summary['redirect_chain'] = [
+            {
+                "key": thing.key,
+                "occurrences": {},
+                "updates": {}
+            } for thing in redirect_chain
+        ]
+        summary['resolved_key'] = redirect_chain[-1].key
+
+        for r in summary['redirect_chain']:
+            olid = r['key'].split('/')[-1][2:-1]
```

Comment:
OPTIONAL: Explain why  `[2:-1]`
```suggestion
            olid = r['key'].split('/')[-1][2:-1]  # 'OL1234x' --> '1234'
```